src/fundus/scraping/common_crawl/pipeline.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import annotations
+
+import gzip
+import os
+import re
+from datetime import datetime
+from functools import lru_cache, partial
+from multiprocessing import Manager
+from multiprocessing.context import TimeoutError
+from multiprocessing.pool import MapResult, Pool, ThreadPool
+from queue import Empty, Queue
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Iterator,
+    List,
+    Literal,
+    Optional,
+    Pattern,
+    Set,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import dill
+import more_itertools
+import requests
+from dateutil.rrule import MONTHLY, rrule
+from tqdm import tqdm
+from typing_extensions import ParamSpec
+
+from fundus.publishers.base_objects import PublisherEnum
+from fundus.scraping.article import Article
+from fundus.scraping.common_crawl.html import CCNewsSource
+from fundus.scraping.common_crawl.scraper import CCNewsScraper
+from fundus.scraping.filter import ExtractionFilter, Requires, URLFilter
+
+_T = TypeVar("_T")
+P = ParamSpec("P")
+
+
+# noinspection PyPep8Naming
+class dill_wrapper(Callable[P, _T]):  # type: ignore[misc]
+    def __init__(self, target: Callable[P, _T]):
+        """Wraps function in dill serialization.
+
+        This is in order to use unpickable functions within multiprocessing.
+
+        Args:
+            target (Callable[P, _T): The function to wrap.
+        """
+        self._serialized_target: bytes = dill.dumps(target)
+
+    @lru_cache
+    def _deserialize(self) -> Callable[P, _T]:
+        return cast(Callable[P, _T], dill.loads(self._serialized_target))
```

Comment:
Yes, but we want to cache only one function, which is the deserialized target. The `cache` hits every time `_deserialize` is called. This is to avoid calling `dill.loads` again and again.

Reply:
Yes, it's just on every call. Nothing, confusing really. It was properly just late ;).